PWGHF/vertexingHF/AliAnalysisTaskSELc2pKs0fromKFP.h

```diff
@@ -0,0 +1,124 @@
+/// lines.
+///
+/// \author Jianhui Zhu <zjh@mail.ccnu.edu.cn>, Central China Normal University & GSI Helmholtz Centre for Heavy Ion Research
+/// \date Apr 26, 2019
+
+/* $Id$ */
+
+#ifndef HomogeneousField
+#define HomogeneousField
+#endif
+
+#include "AliAnalysisTaskSE.h"
+#include "AliAODMCParticle.h"
+#include "AliRDHFCutsKFP.h"
+#include "AliNormalizationCounter.h"
+#include "THnSparse.h"
+#include "AliPIDResponse.h"
+#include "AliAODInputHandler.h"
+
+// includes added to play with KFParticle
+#include <vector>
+#include "KFParticleBase.h"
+#include "KFParticle.h"
+#include "KFPTrack.h"
+#include "KFPVertex.h"
+#include "KFVertex.h"
+
+class AliAnalysisTaskSELc2pKs0fromKFP : public AliAnalysisTaskSE  
+{
+    public:
+                                AliAnalysisTaskSELc2pKs0fromKFP();
+                                AliAnalysisTaskSELc2pKs0fromKFP(const char *name, AliRDHFCutsKFP* cuts);
+        virtual                 ~AliAnalysisTaskSELc2pKs0fromKFP();
+
+        virtual void            UserCreateOutputObjects();
+        virtual void            Init();
+        virtual void            LocalInit() {Init();} 
+        virtual void            UserExec(Option_t* option);
+        virtual void            Terminate(Option_t* option);
+
+        void                    SetMC(Bool_t IsMC) {fIsMC=IsMC;}
+        void                    SelectTrack(AliVEvent *event, Int_t trkEntries, Int_t &nSeleTrks, Bool_t *seleFlags);
+        Bool_t                  MakeMCAnalysis(TClonesArray *mcArray);
+        void                    MakeAnaLcFromCascadeHF(TClonesArray *arrayLc2pKs0, AliAODEvent *aodEvent, TClonesArray *mcArray, KFParticle PV);
+        Double_t                InvMassV0atPV(AliAODTrack *trk1, AliAODTrack *trk2, Int_t pdg1, Int_t pdg2);
+        Double_t                CosPointingAngleKF(KFParticle kfp, KFParticle kfpmother);
+        Double_t                CosThetaStarKF(Int_t ip, UInt_t pdgvtx, UInt_t pdgprong0, UInt_t pdgprong1, KFParticle kfpvtx, KFParticle kfpprong0, KFParticle kfpprong1);
+        Bool_t                  CheckVertexCov(AliAODVertex *vtx);
+        Bool_t                  CheckTrackCov(AliAODTrack *track);
+        Bool_t                  CheckKFParticleCov(KFParticle kfp);
+        KFParticle              CreateKFTrack(Double_t *param, Double_t *cov, Float_t Chi2perNDF, Int_t charge, Int_t pdg);
+        KFVertex                CreateKFVertex(Double_t *param, Double_t *cov);
+        KFParticle              CreateKFParticleFromAODtrack(AliAODTrack *track, Int_t pdg);
+        KFParticle              CreateKFMotherParticle(AliAODTrack *track1, AliAODTrack *track2, Int_t pdg1, Int_t pdg2);
+        KFParticle              CreateSecKFParticle(KFParticle kfp1, AliAODTrack *track2, Int_t pdg1, Int_t pdg2);
+        Double_t                LdL(KFParticle kfpParticle, KFParticle PV);
+        Int_t                   MatchToMCKs0(AliAODTrack *v0Pos, AliAODTrack *v0Neg, TClonesArray *mcArray);
```

Comment:
Is there a reason for not using the `MatchToMC` function defined in https://github.com/alisw/AliPhysics/blob/62aec2a74e94bcdf3408f4d72f0149a9af008a73/PWGHF/vertexingHF/AliAODRecoCascadeHF.h#L107?